.github/workflows/ci.yml

```diff
@@ -53,14 +53,15 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # TODO: add windows which does not support container
-        os: [ ubuntu-22.04 ]
+        os:
+          - ubuntu-22.04
+          - windows-2022
     runs-on: ${{ matrix.os }}
-    container:
-      image: xd009642/tarpaulin:0.30.0
-      options: --security-opt seccomp=unconfined
     steps:
       - uses: actions/checkout@v4
+      - uses: cargo-bins/cargo-binstall@main
+      - name: Install tarpaulin
+        run: cargo binstall -y cargo-tarpaulin
```

Comment:
@crrow using `cargo-bins/cargo-binstall` works! The job fails due to some test failure - can you look into and fix any code if needed? meanwhile i'll get binstall whitelist so it can be triggered here.

Reply:
@xushiyan I'd love to, but I'd prefer to address the Windows code in a separate pull request.

Reply:
What's your thought on separator and test case? Some test failed because of rust's automatic normalize, and URL parse failed since some test cases are Linux path, and it cannot be parsed on Windows.

By the way, I think some test cases can be rewritten with https://crates.io/crates/test-case

Reply:
@crrow absolutely! we should separate test code fix from CI setup change. Feel free to create another PR for the test fix wrt url parsing. You may also make the code change here so we know the test fix can pass with the CI, then separate it out. It's up to you.

I've thought about using some test helper lib but not sure if https://github.com/frondeus/test-case is the best candidate. there are many crates maintained by community like https://github.com/la10736/rstest seems more popular with more recent releases. I'll create an issue about this for future refactoring work.

Reply:
@xushiyan Do we have to handle this [case?](https://github.com/apache/hudi-rs/blob/main/crates/core/src/storage/utils.rs#L94) , its tricky for windows

Reply:
> @xushiyan Do we have to handle this [case?](https://github.com/apache/hudi-rs/blob/main/crates/core/src/storage/utils.rs#L94) , its tricky for windows

@crrow that path is a unix path which should not be applicable to windows os. Can we split that case to test case which does not run on windows, and also add a windows specific test case with an example windows path ?

Reply:
Sure, I'd love to

Reply:
Oh...I found that its still tricky, like "C:\\foo", `url::parse` will success, it thoughts 'c' is the schema, rather than a filepath; for "\\foo", `url::parse` and `url::from_filepath` will all fail; We can handle both cases by adding additional check like if current os is windows and the given str has ":\\".

Can we be stricter about what qualifies as a 'valid URL'? Like a valid URL must have a schema.

Reply:
@crrow would you do a bit of research on how windows paths are usually handled in consideration of scheme? we want to be flexible in supporting paths - it can start with `C:\`, `file:///`, `/`, `s3://` as these are valid paths anyway.